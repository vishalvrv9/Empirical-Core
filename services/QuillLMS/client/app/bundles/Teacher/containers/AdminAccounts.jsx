'use strict'
import Resource from '../components/cms/resources/resource.jsx'
import Cms from './Cms.jsx'
import React from 'react'

export default React.createClass({

  resourceComponentGenerator: function (cmsComponent) {

    var initialModel = {
      id: null,
      name: null,
      admins: []
    };

    var savingKeys = ['id', 'name', 'admins', 'teachers'];
    var fieldsToNormalize = [];

    var formFields = [
      {
        name: 'name'
      }
    ];

    var nestedResources = [
      {
        name: 'admins',
        message: 'must be an existing admin (can create one in the Admin cms)',
        formFields: [{name: 'email'}],
        identifier: 'email',
        findOrCreate: 'find'
      },
      {
        name: 'teachers',
        message: 'must be an existing teacher',
        formFields: [{name: 'email'}],
        identifier: 'email',
        findOrCreate: 'find'
      }
    ]



    return (
      <div>
        <div className='row'>
          <div className='col-xs-12'>
            <Resource
               resourceNameSingular='admin_account'
               resourceNamePlural='admin_accounts'
               initialModel={initialModel}
               resource={cmsComponent.state.resourceToEdit}
               formFields={formFields}
               savingKeys={savingKeys}
               fieldsToNormalize={fieldsToNormalize}
               nestedResources={nestedResources}
               returnToIndex={cmsComponent.returnToIndex} />
          </div>
        </div>
      </div>
           );
  },

  render: function () {
    return (
      <Cms resourceNameSingular='admin_account'
              resourceNamePlural='admin_accounts'
              resourceComponentGenerator={this.resourceComponentGenerator}/>

    );
  }
});
