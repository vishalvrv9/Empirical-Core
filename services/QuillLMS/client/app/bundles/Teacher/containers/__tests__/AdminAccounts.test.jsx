import React from 'react';
import { shallow } from 'enzyme';

import AdminAccounts from '../AdminAccounts.jsx';

describe('AdminAccounts container', () => {

  it('should render', () => {
    expect(shallow(<AdminAccounts />)).toMatchSnapshot();
  });

});
