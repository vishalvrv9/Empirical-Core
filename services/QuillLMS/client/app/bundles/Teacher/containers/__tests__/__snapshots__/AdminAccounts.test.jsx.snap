// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminAccounts container should render 1`] = `
<Cms
  resourceComponentGenerator={[Function]}
  resourceNamePlural="admin_accounts"
  resourceNameSingular="admin_account"
/>
`;
